rainier-core/src/main/scala/com/stripe/rainier/core/Predictor.scala

```diff
@@ -5,66 +5,66 @@ import com.stripe.rainier.compute._
 /**
   * Predictor class, for fitting data with covariates
   */
-trait Predictor[X, Y] extends Likelihood[(X, Y)] {
-  private[core] type Q
-  private[core] type R
-  type P = (Q, R)
-  private[core] def create(q: Q): Distribution.Aux[Y, R]
-  private[core] def xq: Wrapping[X, Q]
+sealed trait Predictor[X, L] { self =>
+  type P
+  protected def encoder: Encoder[X] { type U = P }
```

Comment:
Would it be equivalent to annotate this as `Encoder.Aux[X, P]`?

Reply:
Yes.